src/program.ts

```diff
@@ -1088,6 +1096,51 @@ export class Program extends DiagnosticEmitter {
     }
   }
 
+  /** Marks virtual members in a base class overloaded in this class. */
+  private markVirtuals(thisPrototype: ClassPrototype, basePrototype: ClassPrototype): void {
+    // TODO: make this work with interfaaces as well
+    var thisInstanceMembers = thisPrototype.instanceMembers;
+    if (thisInstanceMembers) {
+      do {
+        let baseInstanceMembers = basePrototype.instanceMembers;
+        if (baseInstanceMembers) {
+          for (let _values = Map_values(thisInstanceMembers), j = 0, l = _values.length; j < l; ++j) {
+            let thisMember = _values[j];
+            if (!thisMember.is(CommonFlags.CONSTRUCTOR) && baseInstanceMembers.has(thisMember.name)) {
```

Comment:
Could we also exclude private methods which guarantee will never overloaded?

Reply:
Yeah, one of these edge cases to look into :)